harness/determined/deploy/gcp/cli.py

```diff
@@ -427,7 +427,7 @@ def parse(s: str) -> Tuple[str, str]:
                         ),
                         Arg(
                             "--preemptible",
-                            type=bool,
+                            type=string_to_bool,
```

Comment:
why is this `string_to_bool` and not `BoolOptArg`?

Reply:
You're right.

Reasoning was that despite its name, `BoolOptArg` args aren't really optional, and aren't appropriate when the system doesn't want defaults. But this is one of those cases where having defaults is appropriate. I'll fix.

Reply:
Well, maybe not. 2 reasons:
1. (minor reason) Switching to `BoolOptArg` changes the interface from `--preemptible=true|false` to `--preemptible|--no-preemptible`, and this is more likely to have user impact.
2. (major reason) Each of these args is a part of an `ArgGroup`, which contains a list of `Args`. `BoolOptArgs` are not `Args` and do not look like `Args` and at this point I think refactoring `BoolOptArgs` or shimming an interface is not worth this PR. I could instead make each `ArgGroup`'s `child_args` list contain a union of `Args` and `BoolOptArgs`, but this feels like asking for trouble in the future if I don't unify the interface. I'm not convinced that it's worth spending a lot of time on `declarative_argparse`'s lesser-used features because I'm not sure what the future of `declarative_argparse` will be.